src/cdk/accordion/accordion.ts

```diff
@@ -28,4 +32,20 @@ export class CdkAccordion {
   get multi(): boolean { return this._multi; }
   set multi(multi: boolean) { this._multi = coerceBooleanProperty(multi); }
   private _multi: boolean = false;
+
+  /** Opens all enabled accordion items in an accordion where multi is enabled. */
+  openAll(): void {
+    this._openCloseAll(true);
+  }
+
+  /** Closes all enabled accordion items in an accordion where multi is enabled. */
+  closeAll(): void {
+    this._openCloseAll(false);
+  }
+
+  private _openCloseAll(expanded: boolean): void {
+    if (this.multi) {
```

Comment:
There are same cases, even for `[multi]="false"`, where you don't care which `CdkAccordionItem` is open, you just want to make sure that all are closed and it would be nice to also cover that use case, but I'm not sure how this could be done without confusing developers.

Reply:
I guess you could do `this.multi || !expanded`? Will have to experiment.

Reply:
That would be a solution, but we could have a "cleaner" solution, if I may use this word.

This would be a good solution for this particular case. But maybe we could find a more generic solution that could be used in other cases and not confuse people to much, `openAll()` won't work in single mode, but we could have `open()` that works for single mode, opening the first item or in multi mode.

If you look at something like the `SelectionModel` for tables, select and chips, the selection has a `getMultipleValuesInSingleSelectionError`, we could use something like that. And `MatSelectionList.selectedOptions` exposes the model for anyone to use. But probably a selection model would be an overkill for this situation, I'm not sure in how many places we could reuse it.

Also, `MatDrawerContainer` container has a similar behavior but it doesn't add the `All` suffix and it uses `@ContentChildren` to keep a list of children and then call open/close on each children, instead of emitting events, here's a snippet from it:

```
  @ContentChildren(MatDrawer) _drawers: QueryList<MatDrawer>;

  /** Calls `open` of both start and end drawers */
  open(): void {
    this._drawers.forEach(drawer => drawer.open());
  }

  /** Calls `close` of both start and end drawers */
  close(): void {
    this._drawers.forEach(drawer => drawer.close());
  }
```
Using `ContentChildren` instead of emitting an event and rename to `open/close` in order to support "multi/single" mode, would be great.


P.S. Thanks for the work you did on this PR!

Reply:
This maybe warrants a separate issue to investigate?